### PR TITLE
Prepare release 0.106.0

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #3183 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,30 @@
...
+- `target allocator`: Fix collector to target allocator connection in clusters with proxy. (#3187)
+  On clusters with global proxy the collector might fail to talk to target allocator
+  because the endpoint is set to `<ta-service-name>:port` and therefore it will go to proxy
+  and request might be forwarded to internet. Clusters with proxy configure `NO_PROXY` to `.svc.cluster.local` so 
+  the calls to this endpoint will not go through the proxy.
+  
+
+### Components
+
+* [OpenTelemetry Collector - v0.106.1](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.106.1)
```
Comment (pavolloffay) [inline CHANGELOG.md:19]: @iblancasa 

> service: Update all metrics to include otelcol_ prefix to ensure consistency across OTLP and Prometheus metrics (https://github.com/open-telemetry/opentelemetry-collector/pull/9759) 

https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.106.0 

Could you please check if it affects the OCP dashboard?

Comment (iblancasa) [reply CHANGELOG.md:19]: All the metrics used for the dashboard included the `otelcol_` prefix before.

**Comment by iblancasa** [comment]
I think we should have https://github.com/open-telemetry/opentelemetry-operator/pull/3197 merged before doing the release.

**Comment by jaronoff97** [comment]
I would like to merge #3200

**Review by iblancasa — APPROVED** [review]

**Comment by pavolloffay** [comment]
ACK I will merge and rebase this PR

File: README.md [hunk 2]
```diff
@@ -739,6 +739,7 @@ The OpenTelemetry Operator _might_ work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes     | Cert-Manager | Prometheus-Operator |
 |------------------------|----------------| ------------ |---------------------|
+| v0.106.0               | v1.23 to v1.30 | v1           | v0.74.0             |
```
Comment (jaronoff97) [inline README.md:742]: [prom-operator is v0.75.0 now](https://github.com/open-telemetry/opentelemetry-operator/blob/main/go.mod#L23)

Comment (pavolloffay) [reply README.md]: done and rebased with #3200

**Review by jaronoff97 — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

**Pull request merged**